apps/condo/domains/news/tasks/publishNewsItemSharing.js

```diff
@@ -0,0 +1,167 @@
+const get = require('lodash/get')
+const { v4 } = require('uuid')
+
+const { fetch } = require('@open-condo/keystone/fetch')
+const { getLogger } = require('@open-condo/keystone/logging')
+const { getSchemaCtx, getById, find } = require('@open-condo/keystone/schema')
+const { createTask } = require('@open-condo/keystone/tasks')
+
+const { CONTEXT_FINISHED_STATUS } = require('@condo/domains/miniapp/constants')
+const { STATUSES } = require('@condo/domains/news/constants/newsItemSharingStatuses')
+const { NewsItemSharing } = require('@condo/domains/news/utils/serverSchema')
+
+
+const logger = getLogger('publishNewsItemSharing')
+
+const DV_SENDER = { dv: 1, sender: { dv: 1, fingerprint: 'publishNewsItemSharing' } }
+
+const DEFAULT_TIMEOUT = 20 * 1000
+
+
+async function _publishNewsItemSharing (newsItem, newsItemSharing, taskId){
+    const { keystone: contextNewsItemSharing } = getSchemaCtx('NewsItemSharing')
+
+    if (!newsItem) {
+        throw new Error('no news item')
+    }
+
+    if (newsItemSharing.status !== STATUSES.SCHEDULED) return
+    await NewsItemSharing.update(contextNewsItemSharing, newsItemSharing.id, {
+        ...DV_SENDER,
+        status: STATUSES.PROCESSING,
+    })
+
+    const { title, body, type, validBefore, createdAt, publishedAt } = newsItem
+    const sharingParams = get(newsItemSharing, 'sharingParams')
+
+    // Check scopes
```

Comment:
Perhaps all these checks should be placed in a try/catch, so that when errors occur we update the status of `NewsItemSharing`.

Because now if we throw an error, the status will remain `processing`, although it should change to `error`